test/powershell/engine/Api/TaskBasedAsyncPowerShellAPI.Tests.ps1

```diff
@@ -0,0 +1,251 @@
+        BeforeAll {
+            $d1 = New-Object -TypeName 'System.Management.Automation.PSDataCollection[int]'
+            $d2 = New-Object -TypeName 'System.Management.Automation.PSDataCollection[int]'
+            foreach ($i in 1..20) {
+                $d1.Add($foreach.Current)
+                $foreach.MoveNext() > $null
+                $d2.Add($foreach.Current)
+            }
+            $d1.Complete()
+            $d2.Complete()
+            $script = "`$input${sbStub}"
+        }
+
+        It 'can invoke multiple scripts asynchronously with input' {
+            $ps1 = [powershell]::Create()
+            $ps2 = [powershell]::Create()
+            try {
+                $r1 = $ps1.AddScript($script).InvokeAsync($d1)
+                $r2 = $ps2.AddScript($script).InvokeAsync($d2)
+                [System.Threading.Tasks.Task]::WaitAll(@($r1, $r2))
+                $r1.Status | Should -Be ([System.Threading.Tasks.TaskStatus]::RanToCompletion)
+                $r1.IsCompletedSuccessfully | Should -Be $true
+                $r2.Status | Should -Be ([System.Threading.Tasks.TaskStatus]::RanToCompletion)
+                $r2.IsCompletedSuccessfully | Should -Be $true
+                $allResults = @($r1.Result) + @($r2.Result)
+                Compare-Object -ReferenceObject @(1..20) -DifferenceObject $allResults.Value -SyncWindow 20 | Should -Be $null
+            } finally {
+                $ps1.Dispose()
+                $ps2.Dispose()
+            }
+        }
+
+        It 'can invoke multiple scripts asynchronously with input and capture output' {
+            $ps1 = [powershell]::Create()
+            $ps2 = [powershell]::Create()
+            try {
+                $o = New-Object -TypeName 'System.Management.Automation.PSDataCollection[PSObject]'
+                $r1 = $ps1.AddScript($script).InvokeAsync($d1, $o)
+                $r2 = $ps2.AddScript($script).InvokeAsync($d2, $o)
+                [System.Threading.Tasks.Task]::WaitAll(@($r1, $r2))
+                $o.Complete()
+                $r1.Status | Should -Be ([System.Threading.Tasks.TaskStatus]::RanToCompletion)
+                $r1.IsCompletedSuccessfully | Should -Be $true
+                $r2.Status | Should -Be ([System.Threading.Tasks.TaskStatus]::RanToCompletion)
+                $r2.IsCompletedSuccessfully | Should -Be $true
+                Compare-Object -ReferenceObject @(1..20) -DifferenceObject $o.Value -SyncWindow 20 | Should -Be $null
+            } finally {
+                $ps1.Dispose()
+                $ps2.Dispose()
+            }
+        }
+    }
+
+    Context 'PowerShell::StopAsync' {
+        It 'can stop multiple scripts that are running asynchronously' {
+            $ps = @([powershell]::Create(),[powershell]::Create())
+            try {
+                $ir = @($ps[0].AddScript("@(1,3,5,7,9,11,13,15,17,19)${sbStub}").InvokeAsync(),
+                        $ps[1].AddScript("@(2,4,6,8,10,12,14,16,18,20)${sbStub}").InvokeAsync())
```

Comment:
Bumped up to cycle through the loop 240 times (2 minutes total). That should be long enough while continuing to use the script stub at the top of the file.

Reply:
Yeah, that will run long enough. But why are you using that script for this test? All you need here is for the async call to last a while so that it can be stopped before finish, so why not use a simple script like `start-sleep -second 60`?

Reply:
Fair point, I took your script instead. The other was simply an artifact of copying another test and modifying it.